src/common/service/NetRemoteService.cxx

```diff
@@ -389,6 +427,76 @@ NetRemoteService::WifiAccessPointDisable([[maybe_unused]] ::grpc::ServerContext*
     return grpc::Status::OK;
 }
 
+using Microsoft::Net::Wifi::Dot11PhyType;
+
+::grpc::Status
+NetRemoteService::WifiAccessPointSetPhyType([[maybe_unused]] ::grpc::ServerContext* context, const ::Microsoft::Net::Remote::Wifi::WifiAccessPointSetPhyTypeRequest* request, ::Microsoft::Net::Remote::Wifi::WifiAccessPointSetPhyTypeResult* response)
+{
+    using Microsoft::Net::Wifi::Ieee80211AccessPointCapabilities;
+
+    LOGD << std::format("Received WifiAccessPointSetPhyType request for access point id {}", request->accesspointid());
+
+    WifiAccessPointOperationStatus status{};
+    status.set_code(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeSucceeded);
+
+    auto handleFailure = [&](WifiAccessPointOperationStatusCode statusCode, std::string statusMessage) {
+        LOGE << statusMessage;
+
+        status.set_code(statusCode);
+        status.set_message(statusMessage);
+
+        response->set_accesspointid(request->accesspointid());
+        *response->mutable_status() = std::move(status);
+
+        return grpc::Status::OK;
+    };
+
+    // Check if PHY type is provided.
+    if (request->phytype() == Dot11PhyType::Dot11PhyTypeUnknown) {
+        return handleFailure(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeInvalidParameter, "No PHY type provided");
+    }
+
+    // Create an AP controller for the requested AP.
+    auto accessPointWeak = m_accessPointManager->GetAccessPoint(request->accesspointid());
+    auto accessPointController = detail::IAccessPointWeakToAccessPointController(accessPointWeak);
+    if (!accessPointController) {
+        return handleFailure(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeAccessPointInvalid, std::format("Failed to create controller for access point {}", request->accesspointid()));
+    }
+
+    // Convert PHY type to Ieee80211 protocol.
+    auto ieeeProtocol = detail::NetRemotePhyTypeToIeeeProtocol(request->phytype());
+
+    // Check if Ieee80211 protocol is supported by AP.
+    try {
+        const auto& supportedIeeeProtocols = accessPointController->GetCapabilities().Protocols;

```

Comment:
This is a dangling reference since the capabilities object doesn't exist after this line completes. So, need to update this to either make a copy of the protocols, or, just keep the capabilities as a local. I'd probably do the latter so we could inspect the full capabilities if debugging.
```suggestion
        auto accessPointCapabilities = accessPointController->GetCapabilities();
        const auto& supportedIeeeProtocols = accessPointCapabilities.Protocols;
```